core/db/migrate/20200117125833_add_merged_to_order_id.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class AddMergedToOrderId < ActiveRecord::Migration[5.2]
+  def change
+    add_column :spree_orders, :merged_to_order_id, :integer, limit: 4
+    add_foreign_key :spree_orders, :spree_orders, column: :merged_to_order_id
```

Comment:
I would also consider adding a `merged_at` timestamp, especially when reconstructing weird edge cases that can be a precious information. In alternative the default order merger should at least add a note mentioning the date of the merge.

Reply:
@elia I think that would be redundant. We have `state_changes` on orders, and since `merged` is a new state, a new record pops up after merging the order:
```ruby
2.6.3 :001 > order2.state_changes
=> [#<Spree::StateChange:0x00007fea97cd4f90
  id: 1,
  name: "order",
  previous_state: "cart",
  stateful_id: 2,
  user_id: 2,
  stateful_type: "Spree::Order",
  next_state: "merged",
  created_at: Fri, 07 Feb 2020 11:28:08 UTC +00:00,
  updated_at: Fri, 07 Feb 2020 11:28:08 UTC +00:00>]
```